pypdf/_cmap.py

```diff
@@ -461,6 +461,7 @@ def compute_space_width(
                 m = 0
                 cpt = 0
                 for x in w:
+                    x = cast(Any, x).get_object()
```

Comment:
Is this really the correct cast? I would have assumed that

```suggestion
                    x = cast(int, x.get_object())
```

would make more sense here?

Reply:
I thought the same initially as well but mypy still complains about the proposed change here with `error: "int" has no attribute "get_object"  [attr-defined]` since the `get_object` method is being called before `x` is cast, i.e. when mypy still thinks `x` is an `int`.

Reply:
Instead of Any, does PdfObject works?

Reply:
Looks like not. Mypy throws `Incompatible types in assignment (expression has type "PdfObject | None", variable has type "int")  [assignment]`

Reply:
The loop variable and the result of `get_object()` need to be two different variables - otherwise *mypy* has issues to re-assign a new type here. (Checked with Python 3.9 and `mypy==1.4.1` as used in CI.)

My local test with

```python3
                for _x in w:
                    x = cast(int, _x.get_object())
```

did not yield any *mypy* warnings.